src/sknnr_spatial/utils/image.py

```diff
@@ -3,17 +3,11 @@
 
 import numpy as np
 import xarray as xr
-from typing_extensions import Any, Concatenate, ParamSpec, TypeVar
+from typing_extensions import Any, Concatenate
 
-from ..image._base import ImagePreprocessor, ImageType, ImageWrapper
-from ..image.dataarray import DataArrayWrapper
-from ..image.dataset import DatasetWrapper
-from ..image.ndarray import NDArrayWrapper
+from ..types import RT, ImageType, P
 from .wrapper import GenericWrapper
 
-RT = TypeVar("RT")
-P = ParamSpec("P")
-
 
 def is_image_type(X: Any) -> bool:
     # Feature array images must have exactly 3 dimensions: (y, x, band) or (band, y, x)
```

Comment:
I guess I don't work with multi-band images in `rasterio` very often, because I was convinced it was the opposite, but you're completely right! I haven't thought through the implications, but we should probably modify `load_swo_ecoplot` and `NDArrayImage` to be consistent with the `rasterio` order, right? I think that should be a quick fix, but probably worth a separate PR.

Good catch!

Reply:
Sounds like a plan.  I think there might be users that store their data as multi-band (I don't often myself) and are used to the `rasterio` API that this would be a good consideration.

Reply:
Agreed, tracking this in #29.